src/NuGet.Clients/NuGet.SolutionRestoreManager/VsSolutionRestoreService.cs

```diff
@@ -374,8 +401,17 @@ private static string[] HandleClear(string[] input)
                     var val = GetPropertyValueOrNull(tfm.Properties, propertyName);
                     return val != null ? valueFactory(val) : default(TValue);
                 })
-                .Distinct()
-                .SingleOrDefault();
+                .Distinct();
+        }
+
+        // Trying to fetch a property value from tfm property bags.
+        // If defined the property should have identical values in all of the occurances.
+        private static TValue GetSingleNonEvaluatedPropertyOrNull<TValue>(
```

Comment:
style: instead of a separate API, can you just use `GetNonEvaluatedPropertyOrNull` with `.SingleOrDefault()` inline at the dest code?